addons/account/models/res_partner_bank.py

```diff
@@ -63,6 +64,22 @@ def _check_allow_out_payment(self):
                 if not self.env.user.has_group('account.group_validate_bank_account'):
                     raise ValidationError(_('You do not have the right to trust or un-trust a bank account.'))
 
+    @api.depends('acc_number')
+    def _compute_alert_duplicate_bank_account(self):
+        """Add a warning if the bank account is already registered with another partner."""
+        duplicate = self.env['res.partner.bank'].search([('acc_number', 'in', self.mapped('acc_number'))], limit=1)
+        for bank in self:
+            bank.duplicate_account_warning = False
+            if not bank.acc_number or not duplicate:
+                continue
+            if bank != duplicate:
```

Comment:
I don't understand the logic of the `limit=1` with this, is it because we only are doing this check in the form view anyway?
Can't we check than `len(self) == 1` before doing the search then?

Reply:
the `limit=1` is still necessary  to handle cases with multiple duplicates, but I removed the `mapped`